packages/tasit-action/src/TasitAction.test.js

```diff
@@ -353,20 +398,27 @@ describe("TasitAction.Contract", () => {
         errorFn();
       };
 
+      txSubscription = simpleStorage.setValue("hello world");
+
       txSubscription.on("error", errorListener);
 
+      txSubscription.on("confirmation", confirmationListener);
+
+      await mineBlocks(provider, 5);
+
       const snapshotId = await createSnapshot(provider);
 
-      await mineBlocks(provider, 15);
+      await mineBlocks(provider, 20);
 
       expect(confirmationFn.called).to.be.true;
 
       await revertFromSnapshot(provider, snapshotId);
 
-      await mineBlocks(provider, 20);
+      // Note: Without that, ethers.provider will keep the same
+      // receipt.confirmations as before snapshot reversion
+      txSubscription.refreshProvider();
 
-      // Broadcast same transaction again (simulate reorg)
-      txSubscription = simpleStorage.setValue("hello world");
+      await mineBlocks(provider, 15);
 
       expect(errorFn.called).to.be.true;
```

Comment:
Does the error function get called on the first new block because the number of confirmations is lower than before?

Reply:
Yes, not always on the first new block because of pollingInterval vs blockTime issue, but the first poll after that  15 new blocks is emitting error event.

Reply:
Got it - okay, I'll add that as a note in the code

Reply:
```suggestion
      // not always on the first new block because of pollingInterval vs blockTime issue
      // but the first poll after that 15 new blocks is emitting error event
      expect(errorFn.called).to.be.true;
```

Reply:
Diff outdated but I've just commited that.

Reply:
Thanks! Resolving this